core/src/block_creation_loop.rs

```diff
@@ -512,6 +511,19 @@ pub fn start_loop(config: BlockCreationLoopConfig) {
                 break;
             }
 
+            // Check if the entire window is skipped.
+            let highest_parent_ready_slot = leader_window_notifier
+                .highest_parent_ready
+                .read()
+                .unwrap()
+                .0;
+            if highest_parent_ready_slot > end_slot {
+                trace!(
+                    "{my_pubkey}: Skipping production of {slot} because highest parent ready slot is {highest_parent_ready_slot} > end slot {end_slot}"
+                );
+                break;
```

Comment:
might be good to add an extra case `skipped_window_behind_parent_ready` to the `BlockCreationLoopMetrics`